packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/simulationservice/interventions/InterventionPolicy.java

```diff
@@ -40,4 +42,44 @@ public InterventionPolicy clone() {
 		}
 		return clone;
 	}
+
+	/**
+	 * Check each intervention this policy contains
+	 * Within all static interventions
+	 *   if there are duplicate keys appliedTo and timestep this will throw an error
+	 *   If any timestep is negative this will throw an error
+	 * */
+	public void validateInterventionPolicy() throws Exception {
+		final Set<String> foo = new HashSet<>();
+		for (int i = 0; i < this.interventions.size(); i++) {
+			final Intervention intervention = this.interventions.get(i);
+			// For each static intervention within the policy:
+			for (int j = 0; j < intervention.getStaticInterventions().size(); j++) {
+				final StaticIntervention staticIntervention = intervention.getStaticInterventions().get(j);
+				// Check for negative timestamps:
+				final Number time = staticIntervention.getTimestep();
+				if (time.doubleValue() < 0) {
+					final String errorMessage = String.format(
+						"The intervention %s has a timestep %s which is less than 0.",
+						this.getName(),
+						time.toString()
+					);
+					throw new Exception(errorMessage);
+				}
+				// Check for duplicate appliedTo timestep pairs:
+				final String key = staticIntervention.getAppliedTo() + staticIntervention.getTimestep().toString();
+				if (foo.contains(key)) {
```

Comment:
You may want a delimiter here, in the event that someone comes along and create a model and interventions like
- beta1 should be 0.5 at time 5 => beta15
- beta should be 0.2 at time 15 => beta15
Which would be invalid in your scheme I thihnk.

Also please pick a better name than "foo" :)

Reply:
foo to `duplicateCheckSet` 
key to include `atTheTime:` in the middle of appliedTo and timestep